src/components/admin/order/realtime/modal/order-by-product/OrderByProductModal.tsx

```diff
@@ -0,0 +1,129 @@
+import { Order } from '@@types/index';
+import { colFlex, rowFlex } from '@styles/flexStyles';
+import styled from '@emotion/styled';
+import _ from 'lodash';
+import { useRecoilValue } from 'recoil';
+import { productsAtom } from '@recoils/atoms';
+import { tabletMediaQuery } from '@styles/globalStyles';
+import AppLabel from '@components/common/label/AppLabel';
+import { Color } from '@resources/colors';
+
+const ModalContainer = styled.div`
+  ${colFlex({ align: 'center' })};
+  position: fixed;
+  top: calc(50% - 275px);
+  right: 150px;
+  background-color: rgba(233, 233, 233, 0.8);
+  border-radius: 10px;
+  width: 240px;
+  height: 550px;
+  z-index: 1999;
+  padding: 25px;
+  box-sizing: border-box;
+  gap: 40px;
+
+  ${tabletMediaQuery} {
+    right: 120px;
+  }
+`;
+
+const HeaderContainer = styled.div`
+  ${colFlex({ justify: 'center', align: 'center' })};
+  width: 100%;
+  gap: 3px;
+`;
+
+const ProductContainer = styled.div`
+  ${colFlex()};
+  width: 100%;
+  height: 450px;
+  overflow-y: auto;
+  background: ${Color.LIGHT_GREY};
+  border-radius: 10px;
+  box-sizing: border-box;
+  padding: 10px;
+  gap: 12px;
+`;
+
+const ProductItem = styled.div`
+  ${rowFlex({ align: 'center' })};
+  gap: 20px;
+`;
+
+const ProductImage = styled.img`
+  width: 50px;
+  height: 50px;
+  border-radius: 50%;
+`;
+
+const ProductInfoContainer = styled.div`
+  ${colFlex({ justify: 'center', align: 'flex-start' })};
+  overflow: hidden;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+  word-break: break-all;
+`;
+
+interface OrderByProductModalProps {
+  orders: Order[];
+  isModalOpen: boolean;
+}
+
+function OrderByProductModal({ orders, isModalOpen }: OrderByProductModalProps) {
+  const products = useRecoilValue(productsAtom);
+
+  if (!isModalOpen) {
+    return null;
+  }
+
+  const productCountMap = new Map(
+    [
+      ...orders
+        .reduce((acc, order) => {
+          order.orderProducts.forEach((orderProduct) => {
+            const productId = orderProduct.productId;
+            const quantity = orderProduct.quantity - orderProduct.servedCount;
+
+            acc.set(productId, (acc.get(productId) || 0) + quantity);
+          });
+          return acc;
+        }, new Map<number, number>())
+        .entries(),
+    ].sort((a, b) => b[1] - a[1]),
+  );
+
+  const productMap = _.keyBy(products, (product) => product.id);
```

Comment:
```suggestion
  const productMap = _.keyBy(products, 'id');

  const productCounts: Record<number, number> = {};
  orders.forEach((order) => {
    order.orderProducts.forEach((orderProduct) => {
      const quantity = orderProduct.quantity - orderProduct.servedCount;
      if (quantity > 0) {
        const productId = orderProduct.productId;
        productCounts[productId] = (productCounts[productId] || 0) + quantity;
      }
    });
  });

  const sortedProducts = Object.entries(productCounts)
    .map(([productId, count]) => ({ productId: Number(productId), count }))
    .sort((a, b) => b.count - a.count);
```
map안에서 사용중인 reduce 구문이 눈에 잘 안들어오는 것 같습니다. 그래서 `productCounts`객체에 {productId, 남은 수량} 형태로 저장했습니다. 이후 Object.entries()를 통해 객체를 배열로 변환해줍니다.

개인적으로 객체를 이용한 방법이 좀 더 가독성이 좋은 것 같습니다. 어떻게 생각하시는지요

Reply:
넵 좋은 것 같네요

그런데 entry가 <number, number>라서 `productId: Number(productId)` 부분을 필요 없어보이는데 쓰신 이유가 있나요?